packages/auth/src/providers/cognito/apis/confirmSignIn.ts

```diff
@@ -69,10 +69,17 @@ export async function confirmSignIn(
 		clientMetaData[KEY_PASSWORDLESS_SIGN_IN_METHOD] = 'MAGIC_LINK';
 	}
 
-	// TODO: implement confirm sign-in with OTP
-
-	const { username, challengeName, signInSession, signInDetails } =
-		signInStore.getState();
+	const {
+		username,
+		challengeName,
+		signInSession,
+		signInDetails,
+		signInMethod,
+	} = signInStore.getState();
+	if (signInMethod) {
+		clientMetaData[KEY_PASSWORDLESS_ACTION] = 'CONFIRM';
+		clientMetaData[KEY_PASSWORDLESS_SIGN_IN_METHOD] = signInMethod;
```

Comment:
Thanks for adding this. In this case the lin 68 and 69 is no longer necessary. In the same time we should also add the signInMethod of `'MAGIC_LINK'` in the `loadMagicLinkSignInState()`